### ISSUE TITLE
Embedded Text not working

### ISSUE THREAD

**Issue by dylansaintjohn** [bambulab/BambuStudio#3179]
### Bambu Studio Version

1.8.2.56

### Where is the application from?

Bambu Lab Official website

### OS version

Mac Ventura 13.3.1 (a)

### Additional system information

Bambu studio is not correctly generating text using the Text shape tool. The text does not embed into the shape per the online instructions.

### Printer

Bambu X1C

### How to reproduce

Generate text on a small object and slice it.

### Actual results

Text is raised above the surface. Never embedded or flush

### Expected results

Text generated should 'merge' with the object and embedded depth should be the amount it's deep into the object.

### Project file & Debug log uploads

[Light Switch Cover- Bambu Bug.3mf.zip](https://github.com/bambulab/BambuStudio/files/13728908/Light.Switch.Cover-.Bambu.Bug.3mf.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

**Comment by zhimin-zeng-bambulab** [comment]
![image](https://github.com/bambulab/BambuStudio/assets/120697589/6fb1bd4a-a6ec-4467-92f9-575f468a2b33)
@dylansaintjohn 
If you want the text to be flush with the surface of the object, you can change the type of text_shape to "modifier".

**Comment by zhimin-zeng-bambulab** [comment]
@dylansaintjohn   Did the above method solve your problem?

**Comment by dylansaintjohn** [comment]
I believe that solved the problem. Thank you.

Dylan St. John
(404) 993-1861

**Comment by LucaUrbinati44** [comment]
I have the same problem. The "modifier" is not working for me. With "modifier" the text does not even appear after slicing.

**Comment by AdrianGarside** [comment]
This still doesn't work for me. I can't embed - does nothing - the text is still on top of the surface and nothing is embedded. I can move the text down and make it a modifier but that also does nothing - there's no impact on the slicing output. This is really frustrating.

**Comment by donhohler** [comment]
Same here. Nothing suggested above works. Please fix this.

**Comment by zhimin-zeng-bambulab** [comment]
Bug reason: The priority of coloring is higher than that of the modifier, so the color modified by the modifier cannot be displayed

**Comment by AdrianGarside** [comment]
> Bug reason: The priority of coloring is higher than that of the modifier, so the color modified by the modifier cannot be displayed

Yes, I discovered that when trying the same thing in Orca. On a standalone object it worked fine in Orca but not in Bambu Studio - the latter still messed up every attempt to embed the text where it worked correctly in Orca.

But on the object I actually wanted it on which was painted in layers, the painting overrode the part (regardless of how I did it - modifier, embedded text, new part with cut out of same size) so the embedded text all showed as if it was there but vanished as soon as you sliced it.

**Comment by L0b0mau** [comment]
This worked in previous versions but now after updating to 1.10.1.50 it does not work any longer, regardles what I try with the text :/

**Comment by claudel2006** [comment]
I saw that bug after opened a new case! I have exactly the same problem with 1.10.2.64

**Comment by precipice7** [comment]
I am able to work around this by adding the text first, BEFORE any coloring. Add text, thickness 0 for flush with object, embedded depth to whatever you need, and change the color to your desired choice. THEN paint whatever colors you need. Works on the newest Bambu Studio version (1.10.2.76)

**Comment by robinehoward** [comment]
Tring all methods. I still get raised text and can no longer achieve the results I had before. want to embed as void and a flush paintable text. now only produces raised

**Comment by quadlawnmowerman-coder** [comment]
issue is back again version 2.5.0.66, text will not imbed and slice, work fine on previous version

**Comment by Trught** [comment]
I can confirm this behavior. It seems that the Paint function conflicts with the Text shape tool’s Embedded Depth setting.

When the object/text is painted, the text no longer respects the embedded depth value and is sliced as raised/on top of the surface instead of being embedded or flush with the model. Without using Paint, the embedded text behavior appears to work as expected.

A workaround is to use Erase all painting on the object. After removing all painting, the Embedded Depth setting starts working correctly again.

So the issue may not be only with the Text tool itself, but specifically with the interaction between Paint and Embedded Depth.

Bambu Studio Version
2.7.1.62

Where is the application from?
Bambu Lab Official website

OS version
Windows 11 (25H2)
